spicy/src/compiler/visitors/validator.cc

```diff
@@ -17,6 +18,115 @@ using util::fmt;
 
 namespace {
 
+// Helper to validate that a type is parseable.
+hilti::Result<hilti::Nothing> isParseableType(const type::unit::item::Field& f, const Type& pt) {
+    if ( pt.isA<type::Bitfield>() )
+        return hilti::Nothing();
+
+    if ( pt.isA<type::Bytes>() ) {
```

Comment:
Yeah, let's leave a rework for later. Covering all the cases is diffiicult, and I'm sure we'll keep adding to this. Would rather refactor once we have most of that in place (because that might inform the structure)